soc/riscv/riscv-privilege/efinix-sapphire/soc.h

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (c) 2023 Efinix Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __RISCV32_EFINIX_SAPPHIRE_SOC_H_
+#define __RISCV32_EFINIX_SAPPHIRE_SOC_H_
+
+#include "soc_common.h"
+#include <zephyr/arch/common/sys_io.h>
+#include <zephyr/devicetree.h>
+
+#ifndef _ASMLANGUAGE
+
+#endif /* _ASMLANGUAGE */
```

Comment:
This file is empty. Is this needed at all?

Reply:
I left it because the `zephyr/include/zephyr/arch/riscv/irq.h` line  26: has ` #include <soc.h>` (dependency)
so removing the file `soc.h` is causing compilation error.

Reply:
@gmarull just because you love `soc.h` ;)